Bottom: 0bfa91a0c9c4053346366559e13968ce55f2ba29
Top:    155c422fef379ac4739eb6e3849fe5ff74884258
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-10-05 14:38:57 -0700

Refresh of TestRunCommandInterpreterAPI

---
diff --git a/lldb/packages/Python/lldbsuite/test/python_api/interpreter/TestRunCommandInterpreterAPI.py b/lldb/packages/Python/lldbsuite/test/python_api/interpreter/TestRunCommandInterpreterAPI.py
index 195adf60442..ede40134fe4 100644
--- a/lldb/packages/Python/lldbsuite/test/python_api/interpreter/TestRunCommandInterpreterAPI.py
+++ b/lldb/packages/Python/lldbsuite/test/python_api/interpreter/TestRunCommandInterpreterAPI.py
@@ -18,13 +18,16 @@ class CommandRunInterpreterLegacyAPICase(TestBase):
         with open(self.stdin_path, 'w') as input_handle:
             input_handle.write("nonexistingcommand\nquit")
 
+        # Python will close the file descriptor if all references
+        # to the filehandle object lapse, so we need to keep one
+        # around.
         self.filehandle = open(self.stdin_path, 'r')
         self.dbg.SetInputFileHandle(self.filehandle, False)
 
         # No need to track the output
-        devnull = open(os.devnull, 'w')
-        self.dbg.SetOutputFileHandle(devnull, False)
-        self.dbg.SetErrorFileHandle (devnull, False)
+        self.devnull = open(os.devnull, 'w')
+        self.dbg.SetOutputFileHandle(self.devnull, False)
+        self.dbg.SetErrorFileHandle (self.devnull, False)
 
     @add_test_categories(['pyapi'])
     def test_run_session_with_error_and_quit(self):
